packages/python/plotly/plotly/express/_core.py

```diff
@@ -1341,46 +1420,102 @@ def build_dataframe(args, constructor):
 
     # Cast data_frame argument to DataFrame (it could be a numpy array, dict etc.)
     df_provided = args["data_frame"] is not None
+    is_pd_like = False
     needs_interchanging = False
-    if df_provided and not isinstance(args["data_frame"], pd.DataFrame):
-        if hasattr(args["data_frame"], "__dataframe__") and version.parse(
-            pd.__version__
-        ) >= version.parse("2.0.2"):
-            import pandas.api.interchange
-
-            df_not_pandas = args["data_frame"]
-            args["data_frame"] = df_not_pandas.__dataframe__()
-            # According interchange protocol: `def column_names(self) -> Iterable[str]:`
-            # so this function can return for example a generator.
-            # The easiest way is to convert `columns` to `pandas.Index` so that the
-            # type is similar to the types in other code branches.
-            columns = pd.Index(args["data_frame"].column_names())
-            needs_interchanging = True
-        elif hasattr(args["data_frame"], "to_pandas"):
-            args["data_frame"] = args["data_frame"].to_pandas()
+    if df_provided:
+
+        if nw.dependencies.is_pandas_like_dataframe(args["data_frame"]):
```

Comment:
@FBruzzesi This `if` statement has gotten a bit unwieldy -- I've been through the logic and I think it's reasonable, but could you add a brief comment explaining the purpose of each case?

For example it was not obvious to me at first glance that `nw.from_native()` returns the original dataframe if `strict=False`, and probably won't be obvious to anyone who's not familiar with Narwhals.

Reply:
That's valid, thanks - I've heard from someone else too that `strict=False,` isn't obvious, so perhaps in `from_native` we should have a `pass_through` argument instead (but keep `strict` around in stable.v1 of course 😇)

Reply:
I'd agree with that, `pass_through` feels like a more intuitive API for that functionality. 

I'm curious what's the use case for the functionality in the first place -- what can a program do with an object that _might_ be a Narwhals dataframe, but _might_ be some other kind of dataframe that Narwhals was not able to wrap? Is it primarily intended for checking whether Narwhals can handle a specific dataframe type? If so, couldn't you just catch the thrown error?

Reply:
> I'm curious what's the use case for the pass-through functionality

First example I can think of in [scikit-lego add_lags](https://github.com/koaning/scikit-lego/blob/9f4d7fa714a2c3bbdf69fe3f03f420e8e6f75e60/sklego/pandas_utils.py#L272), we allow for numpy arrays or dataframes, and use native methods via separate paths if the result from `from_native` is a narwhals object or a numpy array